### ISSUE TITLE
[docs]: Extension of docs around use cases of YB. 

### ISSUE THREAD

**Issue by hackintoshrao** [yugabyte/yugabyte-db#811]
The following aspects are not very clear from the docs in the site, 

- When not to use YB DB. The kind of use cases and workloads for which YB DB is not a good choice.
   For example does it fit ell when most of your queries are aggregation/grouing for analytical and reporting workloads and so on..
- More use cases for which it is apt to use YB DB. For example, Aggregation of multiple data stores like Redis, Cassandra and Postgres into one system and so on...

**Comment by schoudhury** [comment]
@hackintoshrao Thanks for the feedback, please keep them coming 👍 

Here's the current documentation in this regard, we will work asap on improving these sections further.
https://docs.yugabyte.com/latest/faq/product/ (the first 2 questions answer your questions above).

Regarding your comment on the "Aggregation of multiple data stores like Redis, Cassandra and Postgres into one system" -- that's actually not a goal for YugaByte DB. Given our underlying distributed document store with Google Spanner-like sharding/replication/transactions, we are not a replacement for the traditional Redis/Cassandra and even Postgres use cases. 

- Redis's primary use case is that of a super-fast in-memory cache (with backing data stored in a different DB). We have an efficient block cache for in-memory serving but are certainly not an in-memory store. So instead of thinking YEDIS API as a Redis replacement, we recommend thinking about YEDIS's Redis compatibility as something that enables transactional key-value apps to be written with rich data structures. 

- Cassandra's primary use case is that of an inexpensive, eventually-consistent distributed database. On the other hand, YCQL's Cassandra compatibility enables transactional flexible-schema apps to be written in a SQL-like language. In the process of doing so, we remove most things operationally complex with a traditional Cassandra deployment such as read repairs, anti entropy, JVM tuning, low data density and more. But replacing Cassandra as-is is not a goal for YCQL.

- Postgres' primary use case is that of a reliable single-node transactional database. On the other hand, YSQL is meant to serve apps that need to take zero-data-loss, always-on writes (sometimes for the same keys) across multiple nodes while increasing read performance (through lower latency & higher  throughput).

**Comment by hackintoshrao** [comment]
Thank you for the info @schoudhury for the info, Here are my thoughts, 

- I believe more awareness is necessary on when would transactional key-value store would be useful. That would enhance the developer experience. Some sample applications could catch the attention of the developer community.

- You might have the challenge of developers comparing the modeling of  YCQL to Cassandra. Does the data modeling follow the highly denormalised approach of Cassandra? It would be nice to have more info around when would it be apt to use YCQL and modeling considerations by taking a few use cases and building applications in wide range of languages, this would help the developer experience go a long way.  This cognitive load for an average developer to get started with YB DB is very high now.

**Comment by allesklarbeidir** [comment]
Hi @hackintoshrao, I agree with you and I asked myself the same questions.
I found a good video on Youtube, where @rkarthik007 talked about an actual use case example and the implementation decisions:

[https://youtu.be/0neiV7Qbwgg?t=1700](https://youtu.be/0neiV7Qbwgg?t=1700)

In this video I also learned that in YCQL it is no problem at all to use Secondary Indexes, which in Cassandra CQL is a controverse topic if you look at discussions at StackOverflow etc. I've read a lot about Cassandra being inefficient with Secondary Indexes but in YugaByte it's just a second table which is maintained automatically through transactions. So this seems to be way more efficient than what I read about Cassandra.

I would also like to ask you guys to make things a bit clearer in this regard.
Especially because YugaByte provides the YCQL API which has it's roots in Cassandras CQL, one tends to think that YCQL behaves like Cassandra - except the JSONB Datatype part (which is really awesome for me coming from MongoDB).

I would for example like to know which parameters in the initiation of a connection to YCQL using the cassandra-driver (nodejs) matter - or what do they do? 
**Do they mean the same thing like for Cassandra?**

**Also, is it as "inefficient" to run IN queries on the primary key as it is in Cassandra?**
(You can read about it in multiple blog posts. The basic statement is that the coordinator has to query many nodes for the query and it can crash etc. The suggested solution is to run all queries in parallel as single WHERE pk = "..." query but for me this doesn't seem to be a nice solution.)

**The exact reason why I ask this is the following:**
I have discovered a time-out issue for IN queries when setting the "queryOptions.consistency" parameter.
Since it works properly when using the cassandra-driver (nodejs) directly but causes a time-out issue when using the "express-cassandra" package, I'm not sure yet where the problem lies.
But it would help to know whether this parameter is used at all or if it has the same role like in Cassandra.

The question regarding the performance of IN queries on primary keys is also important to me, because I want to implement Mongoose's "[_.populate()_](https://mongoosejs.com/docs/populate.html)" feature in my library "[Cadoose](https://github.com/allesklarbeidir/cadoose)" to make YCQL even easier to use for people who come from MongoDB and Mongoose.

Thanks!

**Comment by rkarthik007** [comment]
This is great feedback @allesklarbeidir!

Maybe we can add your questions to our FAQ. cc @schoudhury @m-iancu @ndeodhar
